### PR TITLE
Update Setup Wizard

### LINKED ISSUE #4023: add facility setting to disable 'guest' browsing
### Observed behavior

Some organizations prefer that their users always be logged in

### Expected behavior

A facility setting can be added, e.g. "Disable guest access".

When this setting is active:

* "Access as guest" is hidden on the log-in page
* When a user visits a page without an active session, they are redirected to the log-in page

### User-facing consequences

Sometimes students are accessing as a guest because it's easier than dealing with a log-in

### Context

k0.10.0

----

cc @jtamiace @rtibbles this seem reasonable?

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

Update setup wizard to six-part workflow and new settings
screen shot 2018-08-06 at 5 07 03 pm
screen shot 2018-08-06 at 5 06 48 pm
screen shot 2018-08-06 at 5 06 38 pm
screen shot 2018-08-06 at 5 06 17 pm
screen shot 2018-08-06 at 5 06 01 pm
screen shot 2018-08-06 at 5 04 05 pm

### Reviewer guidance

1. Compare visual design and copy with [prototype](https://www.figma.com/file/CiaGXfI6OMOjBF9e8hr1c1Ap/Onboarding-0.11-updates?node-id=111%3A74)
1. Check defaults for steps 3-5 with [these user stories](https://docs.google.com/document/d/1U_8bDJf7STIQXhYtjaUyZmlp7_06ps-9Y_2Ez-Wh7jY/edit)
1. Go forward through as many combinations of settings as possible check to see that post-provisioned app is configured properly.
1. Go backward and forward through setup wizard to make sure info is maintained
1. Make sure keyboard-only control works.

### References

Fixes #4023
…

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Review by mrpau-richard — APPROVED** [review]
I tested this PR @jonboiser. All listed in the Reviewer guidance work with this PR.

File: kolibri/plugins/learn/assets/src/modules/examViewer/actions.js [hunk 1]
```diff
@@ -56,11 +55,9 @@ export function setAndSaveCurrentExamAttemptLog(
           calcQuestionsAnswered(store.rootState.examAttemptLogs)
         );
         // Add this attempt log to the Collection for future caching.
-        const examAttemptLogCollection = ExamAttemptLogResource.fetchCollection({
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/modules/examViewer/actions.js:59]: Was this a part of the resource refactor? Is this currently broken?

File: kolibri/plugins/setup_wizard/assets/src/modules/pluginModule.js [hunk 2]
```diff
@@ -6,24 +6,39 @@ import { permissionPresets } from '../constants';
 export default {
   state: {
     onboardingData: {
+      // Set in DefaultLanguageForm
       language_id: currentLanguage,
+      // Set in FacilityPermissionsForm
       facility: {
         name: '',
       },
+      preset: findKey(permissionPresets, preset => preset.default) || '',
```
Comment (DXCanas) [inline kolibri/plugins/setup_wizard/assets/src/modules/pluginModule.js]: Any reason to use `findkey` over native `someObject[somekey]`?

File: kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/FacilityNameForm.vue [hunk 3]
```diff
@@ -1,11 +1,6 @@
 <template>
 
-  <OnboardingForm
-    :header="$tr('facilityNamingFormHeader')"
-    :description="$tr('facilityNamingFormDescription')"
-    :submitText="submitText"
-    @submit="setFacilityName"
-  >
+  <div>
```
Comment (DXCanas) [inline kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/FacilityNameForm.vue:3]: Might be best to stop calling this component a "form"

File: kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/SuperuserCredentialsForm.vue [hunk 4]
```diff
@@ -149,29 +158,31 @@
...
-          !this.passwordIsInvalid &&
-          !this.passwordConfirmIsInvalid &&
-          !this.facilityIsInvalid
-        );
+        return !this.usernameIsInvalid && !this.passwordIsInvalid && !this.passwordConfirmIsInvalid;
       },
     },
+    beforeDestroy() {
+      // saves data if going backwards in wizard
+      this.saveSuperuserCredentials(false);
```
Comment (DXCanas) [inline kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/SuperuserCredentialsForm.vue:166]: Hmm. Works well, but double submission on "finish". Not sure that's a problem at all, but having a "submit" call might be redundant. 

Not exactly semantic, but what do you think of using submit _only_ as a "next step" and use `onDestroy` for all the forms? Would retain data across every step - that's not gherk'ed out behavior, but might be nice?

**Review by DXCanas — COMMENTED** [review]
- [ ] "More Languages" button not working.


More appropriate for a later bug:
- Keyboard highlight on "back" button at top of form is off

**Review by DXCanas — CHANGES_REQUESTED** [review]
Missed this on last review ^

File: kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/FacilityNameForm.vue (hunk 3 above)
Comment (jonboiser) [reply kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/FacilityNameForm.vue:3]: Maybe `FacilityNameField`?

File: kolibri/plugins/learn/assets/src/modules/examViewer/actions.js (hunk 1 above)
Comment (jonboiser) [reply kolibri/plugins/learn/assets/src/modules/examViewer/actions.js:59]: I might have copied the code over wrong. Basically, you can't call `.set` on the return of `fetchCollection`.

**Comment by jonboiser** [comment]
@DXCanas I think the change language button is broken because all of the locale files are outdated after renaming things (see #4152 ).

Not sure how to fix the keyboard focus thing. The only CSS rules I see for it are in CoreMenuOption.vue.

1. Renamed FacilityNameForm -> FacilityNameTextbox
1. Simplified the initialization of the the `preset` field in pluginModule

File: kolibri/plugins/learn/assets/src/modules/examViewer/actions.js (hunk 1 above)
Comment (DXCanas) [reply kolibri/plugins/learn/assets/src/modules/examViewer/actions.js]: Wait so does this need any updates?

**Review by DXCanas — APPROVED** [review]

**Pull request merged**